userspace/Build.mk

```diff
@@ -279,42 +271,68 @@ userspace/$(APP)/$(BOARD)/doc: sandbox_setup build/gitlongtag
 .PHONY: userspace/$(APP)/$(BOARD)/localtests
 userspace/$(APP)/$(BOARD)/localtests:
 
-.PHONY: userspace/$(APP)/$(BOARD)/program
-userspace/$(APP)/$(BOARD)/program: \
+endef
+
+# ------------------------------------------------------------------------------
+# Macro to define targets for a specific board-app-and-image combination.
+# Arguments:
+# - $(BOARD)
+# - $(APP)
+# - $(IMAGE) [optional suffix to build A/B images]
+define RUST_APP_BOARD_IMAGE_TARGETS
```

Comment:
IMO there ought to be a comment attached to the closing `endef` for this `define` that makes it easy for the reader to visually discern the entirety of the macro body.

Reply:
Done throughout.